aws-xray-recorder-sdk-sql/src/main/java/com/amazonaws/xray/sql/TracingConnection.java

```diff
@@ -0,0 +1,37 @@
+package com.amazonaws.xray.sql;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.sql.Connection;
+import java.sql.Statement;
+
+class TracingConnection {
+    static Connection decorate(Connection c) {
+        return (Connection) Proxy.newProxyInstance(TracingConnection.class.getClassLoader(),
+                new Class[] { Connection.class },
+                new TracingConnectionHandler(c));
+    }
+
+    private static class TracingConnectionHandler implements InvocationHandler {
+
+        private final Connection original;
+
+        TracingConnectionHandler(Connection original) {
+            this.original = original;
+        }
+
+        public Object invoke(Object proxy, Method method, Object[] args)
+                throws IllegalAccessException, IllegalArgumentException,
+                InvocationTargetException {
+            //intercepted methods taken from https://github.com/aws/aws-xray-sdk-java/blob/master/aws-xray-recorder-sdk-sql-mysql/src/main/java/com/amazonaws/xray/sql/mysql/TracingInterceptor.java
+            if (method.getName().equals("prepareCall") || method.getName().equals("prepareStatement") || method.getName().equals("createStatement")) {
```

Comment:
Can we have static final String variables instead of hardcoded strings?

Reply:
can copy the "isStatement(method)" from TracingInterceptor.java

Reply:
Have extracted to isStatement method with static vars (and also isGetConnection and isNewStatement)

TracingInterceptor.isStatement uses a very complicated array logic when a simple "if" is way clearer